From: Nia Espera <a5b6@riseup.net>
Date: Sat, 24 Sep 2022 22:12:05 +0200
Subject: drm/panel: sofef00: remove checks for existence of reset gpio pin

Removes now-useless verification for reset pin on sofef00 panel driver
since s6e3fc2x01 panel has a dedicated driver and reset functionality

Signed-off-by: Nia Espera <a5b6@riseup.net>
Signed-off-by: Caleb Connolly <caleb@connolly.tech>
---
 drivers/gpu/drm/panel/panel-samsung-sofef00.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-samsung-sofef00.c b/drivers/gpu/drm/panel/panel-samsung-sofef00.c
index 68e58b9..0f24e72 100644
--- a/drivers/gpu/drm/panel/panel-samsung-sofef00.c
+++ b/drivers/gpu/drm/panel/panel-samsung-sofef00.c
@@ -44,8 +44,6 @@ struct sofef00_panel *to_sofef00_panel(struct drm_panel *panel)
 
 static void sofef00_panel_reset(struct sofef00_panel *ctx)
 {
-	if (!ctx->reset_gpio)
-		return;
 	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
 	usleep_range(5000, 6000);
 	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
@@ -139,8 +137,7 @@ static int sofef00_panel_prepare(struct drm_panel *panel)
 	ret = sofef00_panel_on(ctx);
 	if (ret < 0) {
 		dev_err(dev, "Failed to initialize panel: %d\n", ret);
-		if (ctx->reset_gpio)
-			gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+		gpiod_set_value_cansleep(ctx->reset_gpio, 1);
 		return ret;
 	}
 
